### ISSUE TITLE
About unbalanced data, dynamical effect and variation of RMSE

### ISSUE THREAD

**Issue by mountain** [pangeo-data/WeatherBench#23]
Hi, buddies,

We all know global meteorological system behave dynamically and statistically. The dynamical variation year by year may lead to statistically unbalanced data. How did we tackle this problem on our benchmark design?

Specifically, I have two question:
1. given a model and parameters trained from past data, how the model perform in future?
2. given a proper model, how the performance of the model vary year by year? the variation reflect the unbalance of the data.

How we judge the unbalance of our dataset. Thanks.

**Comment by raspstephan** [comment]
Hi, great question. 

Yes, weather fluctuates, so we need to make sure that we have a long enough testing period to get statistically significant results. We chose two years for testing. From our experience, two years of data are more than enough for global forecasts. 

A second non-stationarity is climate. However, for average weather forecasting scores this shouldn't matter much.

You are right that it would be informative to plot how the scores vary year by year. However, this isn't that easy to do. I cannot use the operational IFS model from the TIGGE archive because the underlying model is changing all the time. I also can't really use my neural networks because I am using most of the years for training. So the best I can show you is how the mean skill of the climatology varies.

Here you see the relative fluctuation of the climatological RMSE for Z500 by year (T850 looks similar). You can see that the relative differences are less than 2%. This supports what I mentioned above that for global forecasts a year is usually statistically significant. 

![image](https://user-images.githubusercontent.com/2786927/86581250-2672d380-bf80-11ea-8fa3-ebc451c0b766.png)

I hope this helps.

**Comment by mountain** [comment]
Thanks for quick response, @raspstephan . We are investigating this right now, and are curious about your position.

As you mentioned, meteorology and climatology are different, we may separate fluctuations into two parts. I agree with this idea. We may make statement like that, meteorologically the validation set is big enough. But if we want to investigate more precisely, we need define what exactly meteorological and climatological forecast mean, time range and spatio resolution both play roles.

More than that, in data-driven mode, the sample size for training is also a problem. How certain our forecast on rare situations is also a related question.

I am certain you guys had already thought about these, so we'd be glad to hear from you.

**Comment by mountain** [comment]
Sorry, maybe I missed your points.

```
the relative fluctuation of the climatological RMSE
```

Do you mean we can already achieve relative differences less than 2% by using a climatological model as a baseline?

Let me think about that ...

**Comment by raspstephan** [comment]
If you are specifically looking at extreme/rare events, what I said earlier might not be true. Since extreme events are, by definition, very rare, much longer testing periods might be required to get robust results.

In fact, we have not really thought about extremes. One reason is that they are not that easy to define. This also means that the required length of the testing period depends on your exact definition of extremes. 

I am very glad you are thinking more about this and am really looking forward to hearing what you find. Let me know if you need further advice.

About the relative fluctuations, here is how I computed them.

![image](https://user-images.githubusercontent.com/2786927/86591770-82465800-bf92-11ea-90c3-af5b64c7173c.png)

So basically, it's the skill of the climatology for each year divided by the average skill of the climatology.

**Comment by mountain** [comment]
Thanks, I will dig into this...

**Comment by mountain** [comment]
Hi, Stephen @raspstephan,

Some update on this topic: we designed an experiment and the first results had been generated and tested, we still need to reverify this results again on a better model.

Basically,  the findings does not reveal the variation of nature, but the variation of data quality, we can give some qualitative measures on how the  data quality or reanalysis technology have been improved during the last 40 years.

**Comment by mountain** [comment]
Sorry for the late , our experiment results shows contradictory which is difficult to interpret, we think about the results for one week but no easy solution was made.

To study whether the data is balanced or not, 
we have designed an experiments to give 3 tests on each of 2 different models.
But the results from the 2 different models did not show conformity with each other.

The three tests:

* Test A: Training by data from 2013 ~ 2016, testing by data from 1980 ~ 2012
* Test B: Training by data from 1980 ~ 1989, testing by data from 1990 ~ 2018
* Test C: Training by data from even years, testing by data from odd years

The two models: 
* UNet model: the 3 tests are uniformly to show the same trend of the RMSE of the predication which is ***improved***
along decades steadily.
* Conv-tt-LSTM model: the 3 tests are uniformly to show the same level of the RMSE of the predication which is ***kept*** steady.

Model play an important role in the tests which make the results hard to interpret. 

Details of the results are as below:

#### A by UNet

![unet-a](https://user-images.githubusercontent.com/851/89253552-8926c000-d64f-11ea-8227-37ebfa33d807.png)

#### B by UNet

![unet-b](https://user-images.githubusercontent.com/851/89253567-90e66480-d64f-11ea-914c-0d4a959b942d.png)

#### C by UNet

![unet-c](https://user-images.githubusercontent.com/851/89253580-9ba0f980-d64f-11ea-9607-7ce281b3d148.png)

#### A by Conv-tt-LSTM

![convlstm-a](https://user-images.githubusercontent.com/851/89253593-a22f7100-d64f-11ea-9ab7-1c3b5ee31f09.png)

#### B by Conv-tt-LSTM

![convlstm-b](https://user-images.githubusercontent.com/851/89253613-aeb3c980-d64f-11ea-94c1-f0be1bddae60.png)

#### C by Conv-tt-LSTM

![convlstm-c](https://user-images.githubusercontent.com/851/89253622-b4a9aa80-d64f-11ea-9f39-f2761578438f.png)

**Comment by mountain** [comment]
I close the issue now

**Comment by raspstephan** [comment]
Thanks for looking into this. Yes, this does indeed appear inconclusive. But I would say that the trend in the data with time is small enough to not change the results too much.
